paasta_tools/utils.py

```diff
@@ -3723,7 +3723,5 @@ def ldap_user_search(
 
 
 def _reorder_docker_volumes(volumes: List[DockerVolume]) -> List[DockerVolume]:
-    deduped = {
-        v["containerPath"].rstrip("/") + v["hostPath"].rstrip("/"): v for v in volumes
-    }.values()
+    deduped = {v["containerPath"].rstrip("/"): v for v in volumes}.values()
```

Comment:
Can you comment on why this is safe to do and the expected behavior before vs after?

Reply:
@Qmando still looking for your input here :)

Reply:
I will add a variation of this as a comment in the code too:

The behavior before is to deduplicate based on containerPath+hostPath as we can clearly see. This means that if you have two mounts with the same container path, eg,`-v /etc/boto_cfg:/etc/boto_cfg -v /etc/foobar:/etc/boto_cfg`, it will end up keeping both of them and the container will fail to launch. This means that it's currently broken in preventing bad duplicate mounts.

The fix means that it now disregards hostPath when doing this, which is the correct behavior.

I don't remember exactly how this error manifests in paasta, but Docker complains to you with `Error response from daemon: Duplicate mount point: /etc/boto_cfg` if you were to do this.

Reply:
well actually I completely removed this change and did
```
                for existing_mount in volume_mounts:
                    if existing_mount.mount_path == "/etc/boto_cfg":
                        volume_mounts.remove(existing_mount)
                        break
```
instead, but I still think this change is more correct behavior.